Automattic/woocommerce-payments
# Add New Task to Enable Payment Methods

Fixes #6779

#### Changes proposed in this Pull Request

* Adds a new task that displays on both the `Overview` and main WooCommerce pages when a user has completed setup, but has not yet enabled any additional payment methods.
* Fixes a bug with the APM setup flow where a notice would appear in some unintentional circumstances (it was intended to only display to PO incomplete users but was displaying to other users too).









#### Testing instructions

* Verify task matches the design in #6779.
* Run `npm run listen` on your server.
* Ensure you have no APMs enabled on your store (you can do this by disabling them in `Payments > Settings`, so that only `Card` is enabled.
* Onboard a new account using Progressive Onboarding.
* You will be taken to the `Overview` page after partial onboarding to reach "PO enabled" status.
* Verify that you cannot see the new task. (Because you aren't yet `PO Complete`, you cannot yet add APMs.
* Complete full onboarding to enable deposits.
* Verify that you see the new task on the `Overview` page. 
* Click on the new task, you should be redirected to the APM setup.
* Enable at least one new Payment Method and complete the APM setup.
* Go back to the `Overview` page and verify the task is now gone.

## The task on the Overview page:

<img width="728" alt="Screenshot 2023-09-05 at 12 30 14" src="https://github.com/Automattic/woocommerce-payments/assets/11770181/f7c77d29-afbb-4990-a9ad-44944b583168">

## The task on WooCommerce Home:

<img width="711" alt="Screenshot 2023-09-05 at 12 30 30" src="https://github.com/Automattic/woocommerce-payments/assets/11770181/37e50a8f-9b2a-4a2d-bbbd-19f5ad323578">


## Implementation Notes

- The task should display in the following conditions:
      - User has completed KYC (submitted their details)
      - User is not in the `PO Enabled` state (they either need to be `PO Complete`, or a normal, non-PO account - we show the task for normal users because they will also be redirected to the Overview page after onboarding.)
      - User has exactly one payment method enabled. 
- Currently, we display the task if exactly 1 payment method is enabled (this would be card only, since card is required). Ideally, I'd like to hide the task after the user has visited the APM setup, but I'm not sure on the best way to handle it. And maybe it is desired the task displays again if the user disables their APMs? (cc. @elizaan36 for thoughts on this 🙂)





-------------------


**Post merge**
